Fixed: #20050 Fix Button Alignment of Send Button in Nft-Details

ducksblock opened the pull request:
## Explanation
The button was not aligned in the NFT details page. I have fixed the issue by assigning it appropriate CSS properties.

* Fixes #20050 

## Screenshots/Screencaps

<!-- If you're making a change to the UI, make sure to capture a screenshot or a short video showing off your work! -->

### Before
![metamask-before](https://github.com/MetaMask/metamask-extension/assets/105146670/2fc8b003-d767-4d22-a3d0-f675bb7ad8af)


<!-- How did the UI you changed look before your changes? Drag your file(s) below this line: -->

### After
![metamask-after-2](https://github.com/MetaMask/metamask-extension/assets/105146670/768bcec6-1d10-4ff1-ad13-c9f7d82b9933)


<!-- How does it look now? Drag your file(s) below this line: -->

## Manual Testing Steps

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to the NFTs Tab
- Go to any NFT detail in the popup view
-->

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

Commit:
Fixed: #20050 Fix Button Alignment of Send Button in Nft-Details

ducksblock commented:
I have read the CLA Document and I hereby sign the CLA

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Fix: #20050 changed the CSS to align the send button

Signed-off-by: ducksblock <pranavjoshi71619@gmail.com>

ducksblock committed:
Merge branch 'develop' of https://github.com/ducksblock/metamask-extension into develop

ducksblock committed:
Reverted back to the original formatting

ducksblock commented:
Hello @NidhiKJha I have changed the CSS properties. Let me know whether this was the expected code change or not.

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

NidhiKJha commented on ui/components/app/nft-details/index.scss:
```diff
@@ -112,7 +112,7 @@ $spacer-break-small: 16px;
   }
 
   &__send-button {
-    margin-inline-end: 8px;
+    margin: 8px;
```
I think you can remove this margin completely here

NidhiKJha replied:
Updating it to 4px still looks broken
![Screenshot 2023-07-25 at 12 59 39 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/6c1082aa-e9f7-4d4e-8c75-459fb14dc9b6)

NidhiKJha commented on ui/pages/asset/asset.scss:
```diff
@@ -28,6 +28,6 @@
 .asset-options {
   &__button {
-    padding: 2px 0 2px 8px;
+    padding: 2px 4px 2px 4px;
```
This change is not required, revert it please

NidhiKJha commented on ui/components/app/nft-details/nft-details.js:
```diff
@@ -298,8 +297,8 @@ export default function NftDetails({ nft }) {
                   >
-                    {`${Number(decWEIToDecETH(lastSale.total_price))} ${
-                      lastSale.payment_token.symbol
+                    {`${Number(decWEIToDecETH(lastSale.total_price))}
+                      ${lastSale.payment_token.symbol
```
Need to revert this as well, it's causing the lint error in CI for your PR

NidhiKJha reviewed (changes requested):
I have left some comments here, you should look at the parent of the send button fpr the fix, the margin is getting added there.

ducksblock committed:
Changes as requested

ducksblock committed:
Merge branch 'develop' into develop

ducksblock commented:
Hey @NidhiKJha first of all I apologize, I am new to open source contributions and working on big repos like this. I am trying my best to understand the code to contribute to Metamask. I have changed the code as per the request and I hope this time it is the correct one, if not then I am down for another try. Thanks!

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
updated changes

ducksblock committed:
Merge branch 'develop' of https://github.com/ducksblock/metamask-extension into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

NidhiKJha commented on ui/components/app/nft-details/nft-details.js:
```diff
@@ -299,9 +298,8 @@ export default function NftDetails({ nft }) {
                   >
-                    {`${Number(decWEIToDecETH(lastSale.total_price))} ${
-                      lastSale.payment_token.symbol
-                    }`}
+                    {`${Number(decWEIToDecETH(lastSale.total_price))} ${lastSale.payment_token.symbol
+                      }`}
```
Need to fix this lint issue here

NidhiKJha commented on ui/components/app/nft-details/nft-details.js:
```diff
@@ -459,4 +458,4 @@ NftDetails.propTypes = {
     }),
   }),
-};
+};
```
Re: lint fix

NidhiKJha commented on ui/components/app/nft-details/nft-details.js:
```diff
@@ -425,6 +423,7 @@ export default function NftDetails({ nft }) {
             color={TextColor.textAlternative}
             variant={TextVariant.bodySm}
             as="h6"
+            marginTop={4}
```
This doesn't fix the alignment here infact it's adding an extra margin top.

NidhiKJha reviewed (commented):
@ducksblock No worries! I have left some comments.

NidhiKJha commented:
These changes doesn't seem to fix the issue, you need to look into the Button to add extra margin
![Screenshot 2023-07-31 at 2 23 16 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/bda3f4d2-d7d4-4906-8888-b866d04670fa)

ducksblock commented on ui/components/app/nft-details/nft-details.js:
```diff
@@ -145,7 +145,6 @@ export default function NftDetails({ nft }) {
       <Box
         display={Display.Flex}
         width={inPopUp ? BlockSize.Full : BlockSize.Half}
-        margin={inPopUp ? [4, 0] : null}
```
@NidhiKJha I removed this and added the marginTop in the other change to make it aligned. I have tested it, it is aligned for me.

NidhiKJha replied:
Aah! My bad. Yes, this should fix this. Thanks @ducksblock. Please fix the lint issues to make the CI checks pass and we are good to go.

ducksblock replied:
@NidhiKJha It'd be great if you can help me fix these lint errors. As I said I am new to this and these lint errors are so annoying haha

ducksblock replied:
@NidhiKJha I have fixed the lint issues, kindly review it. Thanks.

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Fixed the lint issue

ducksblock committed:
Merge branch 'develop' of https://github.com/ducksblock/metamask-extension into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
lint error fix

ducksblock committed:
Merge branch 'develop' into develop

NidhiKJha reviewed (dismissed):
The changes look good to me. Thanks for your contribution @ducksblock.

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

NidhiKJha commented:
Hey @ducksblock we still have a snapshot failure. Can you please fix this. You can use this command `yarn jest --updateSnapshot`

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
jest fix

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock commented:
> Hey @ducksblock we still have a snapshot failure. Can you please fix this. You can use this command `yarn jest --updateSnapshot`

@NidhiKJha I have fixed the issues.

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock commented:
@NidhiKJha Hi, I would like to know if this issue has anything remaining on my side or will it get merged soon?

ducksblock committed:
Merge branch 'develop' into develop

vthomas13 committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

vthomas13 committed:
Merge branch 'develop' into develop

darkwing committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

NidhiKJha commented:
Hello @ducksblock to merge this PR, all the CI checks need to pass. I think there is a conflict, please refer to this [comment ](https://github.com/MetaMask/metamask-extension/pull/20113#issuecomment-1643910295)by CLA bot. 

![Screenshot 2023-08-21 at 2 36 52 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/bfc28e0e-b7eb-43ca-9169-79f2aa8a8f40)

ducksblock commented:
I have read the CLA Document and I hereby sign the CLA

ducksblock commented:
I have read the CLA Document and I hereby sign the CLA

ducksblock commented:
> Hello @ducksblock to merge this PR, all the CI checks need to pass. I think there is a conflict, please refer to this [comment ](https://github.com/MetaMask/metamask-extension/pull/20113#issuecomment-1643910295)by CLA bot.
> 
> ![Screenshot 2023-08-21 at 2 36 52 PM](https://user-images.githubusercontent.com/39872794/261975374-bfc28e0e-b7eb-43ca-9169-79f2aa8a8f40.jpg)

I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?

NidhiKJha commented:
> I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?

@ducksblock I think it's best to follow the guidelines to update the email as mentioned in the CLA comment. Hopefully that should fix this issue

ducksblock commented:
> > I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?
> 
> @ducksblock I think it's best to follow the guidelines to update the email as mentioned in the CLA comment. Hopefully that should fix this issue

I have updated my email as mentioned in the comment.

ducksblock committed:
Merge branch 'develop' into develop

NidhiKJha commented:
> > > I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?
> > 
> > 
> > @ducksblock I think it's best to follow the guidelines to update the email as mentioned in the CLA comment. Hopefully that should fix this issue
> 
> I have updated my email as mentioned in the comment.

Since you updated the email can you resign it by commenting `I have read the CLA Document and I hereby sign the CLA`

ducksblock commented:
I have read the CLA Document and I hereby sign the CLA

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'MetaMask:develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock commented:
> I have read the CLA Document and I hereby sign the CLA

@NidhiKJha This is not working. Any suggestions are much appreciated.

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

ducksblock committed:
Merge branch 'develop' into develop

brad-decker commented:
<img width="235" alt="Screenshot 2023-09-22 at 11 26 03 AM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/b0440f4e-3c60-4d48-85af-54f8fa5d662d">
Did you happen to commit your last commit on a different machine? it has your name but isn't pulling your profile which might indicate that it thinks that last commit is a new author. You could try to ensure you're logged into your proper machine, your git user name and email is set to the github ones you use, then rebase and force push.

ducksblock committed:
Merge branch 'develop' into develop
